packages/data/lib/repositories/movies_database_repository_impl.dart

```diff
@@ -0,0 +1,129 @@
+import 'package:data/database/dao/genre_dao.dart';
+import 'package:data/database/dao/movie_character_dao.dart';
+import 'package:data/database/dao/movie_dao.dart';
+import 'package:data/database/entities/genre.dart';
+import 'package:data/database/entities/movie.dart';
+import 'package:data/database/entities/movie_character.dart';
+import 'package:domain/entities/base_movie_entity.dart';
+import 'package:domain/entities/movie_character_entity.dart';
+import 'package:domain/repositories/movies_database_repository.dart';
+
+class MoviesDatabaseRepositoryImpl implements MoviesDatabaseRepository {
+  final MovieDao _movieDao;
+  final GenreDao _genreDao;
+  final MovieCharacterDao _movieCharacterDao;
+
+  MoviesDatabaseRepositoryImpl(
+    this._movieDao,
+    this._genreDao,
+    this._movieCharacterDao,
+  );
+
+  @override
+  Future<List<BaseMovieEntity>> getMovies(MovieType movieType) async {
+    final movies = await _movieDao.findMoviesByType(movieType.index);
+
+    final movieEntities = movies.map((movie) async {
+      final genres = await _genreDao.findMovieGenres(movie.id!);
+
+      final genreNames = genres.map((genre) => genre.name).toList();
+
+      return BaseMovieEntity(
+        movie.title,
+        rating: movie.rating,
+        genres: genreNames,
+        runtime: movie.runtime,
+        certification: movie.certification,
+        overview: movie.overview,
+        traktId: movie.traktId,
+        imdbId: movie.imdbId,
+        tmdbId: movie.tmdbId,
+      );
+    });
+
+    return Future.wait(movieEntities);
```

Comment:
seems that u can do it in one query via GROUP_CONCAT

Reply:
@sergei-mikhailovskii-idf  here I need simple take list of genres not concat them as genres either are concated on details page or is taken on genre on home page.